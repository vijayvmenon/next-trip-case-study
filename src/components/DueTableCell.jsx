import React from 'react';
import PropTypes from 'prop-types';
import { createUseStyles } from 'react-jss';
import { ReactComponent as BroadCastIcon } from '../assets/broadcast.svg';

const useStyles = createUseStyles({
  root: {
    position: 'relative',
    display: 'flex',
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  blinker: {
    paddingRight: 3,
    animationName: '$blinker',
    animationDuration: '1s',
    animationTimingFunction: 'cubic-bezier(.5, 0, 1, 1)',
    animationDirection: 'alternate',
    animationIterationCount: 'infinite',
  },
  '@keyframes blinker': {
    to: { opacity: 0 },
  },
});
export default function DueTableCell({ value, row }) {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      {row.actual && <BroadCastIcon className={classes.blinker} />}
      <div>{value}</div>
    </div>
  );
}

DueTableCell.propTypes = {
  value: PropTypes.string.isRequired,
  row: PropTypes.objectOf(PropTypes.any).isRequired,
};
