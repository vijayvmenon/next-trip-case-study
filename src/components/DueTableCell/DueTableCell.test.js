import { render } from '@testing-library/react';
import DueTableCell from './DueTableCell';
import renderer from 'react-test-renderer';

const props = {
  value: '',
  row: { actual: true },
};

describe('Component:DueTableCell.jsx', () => {
  test('DueTableCell.jsx renders correctly', () => {
    const dueTableCell = renderer.create(<DueTableCell {...props} />).toJSON();
    expect(dueTableCell).toMatchSnapshot();
  });

  test('BroadCast Icon is shown when actual prop is true', () => {
    const { getByTestId } = render(<DueTableCell {...props} />);
    expect(getByTestId('broadcast-svg')).toBeTruthy();
  });
});
