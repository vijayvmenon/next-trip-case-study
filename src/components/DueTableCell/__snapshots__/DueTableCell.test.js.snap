// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:DueTableCell.jsx DueTableCell.jsx renders correctly 1`] = `
<div
  className="root-0-2-1"
>
  <svg
    className="blinker-0-2-2"
    data-testid="broadcast-svg"
  >
    broadcast.svg
  </svg>
  <div>
    
  </div>
</div>
`;
